src/apps/settings/src/app/views/Bynder.tsx

```diff
@@ -0,0 +1,237 @@
+import { useEffect, useState } from "react";
+import { ThemeProvider, Box, Stack, Typography, Button } from "@mui/material";
+import PersonRemoveRoundedIcon from "@mui/icons-material/PersonRemoveRounded";
+import { theme } from "@zesty-io/material";
+import { Modal, Login } from "@bynder/compact-view";
+
+import bynderPreview from "../../../../../../public/images/bynder-preview.png";
+import bynderLogo from "../../../../../../public/images/bynder-logo.svg";
+import {
+  useCreateInstanceSettingsMutation,
+  useGetInstanceSettingsQuery,
+  useUpdateInstanceSettingMutation,
+} from "../../../../../shell/services/instance";
+
+// NOTE: cvrt is the bynder refresh token, determines if user is logged in or not
+export const Bynder = () => {
+  const [isLoginOpen, setIsLoginOpen] = useState(false);
+  const [isBynderSessionValid, setIsBynderSessionValid] = useState(false);
+  const [bynderSessionUrl, setBynderSessionUrl] = useState("");
+  const [createInstanceSetting] = useCreateInstanceSettingsMutation();
+  const [updateInstanceSetting] = useUpdateInstanceSettingMutation();
+  const { data: rawInstanceSettings } = useGetInstanceSettingsQuery();
+  const [tokenInterval, setTokenInterval] = useState<NodeJS.Timer>();
+  const [portalUrlInterval, setPortalUrlInterval] = useState<NodeJS.Timer>();
+
+  const bynderPortalUrlSetting = rawInstanceSettings?.find(
+    (setting) => setting.key === "bynder_portal_url"
+  );
+  const bynderTokenSetting = rawInstanceSettings?.find(
+    (setting) => setting.key === "bynder_token"
+  );
+
+  const updateBynderPortalUrl = (url: string) => {
+    if (bynderPortalUrlSetting) {
+      updateInstanceSetting({
+        ...bynderPortalUrlSetting,
+        value: url,
+      });
+    } else {
+      createInstanceSetting({
+        category: "bynder",
+        key: "bynder_portal_url",
+        keyFriendly: "Bynder Portal URL",
+        value: url,
+        dataType: "text",
+      });
+    }
+
+    localStorage.setItem("cvad", url);
```

Comment:
This is because when `updateBynderPortalUrl` is called to either clear out the url or sync it with whatever is saved on the instance, I also need to update that value on localstorage. I could move this out of the function and just place it after `updateBynderPortalUrl` is ran to achieve the same effect.

Reply:
@agalin920 Bynder's UCV component checks the values from localStorage. I tried passing in the token stored on the db to the component directly but it fails to authenticate since based on my understanding of their docs this would only work if we give it a token the came from their [token endpoint](https://bynder.docs.apiary.io/#reference/oauth-2.0/token-endpoint/using-an-authorization-code) which we need to do ourselves. I'm guessing the token that the UCV generates when you log in to it is different from that token that can be directly passed into the component to bypass the whole login thing. The UCV component only works when the token from the db is stored to local storage as "cvrt" and the portal address as "cvad".